src/tensor_product_basis.jl

```diff
@@ -0,0 +1,68 @@
+abstract type TensorProductBasis <: Function end
+
+##Chebyshev basis of the form [T_0(x), T_1(x), ..., T_n(x)] where T is the Chebyshev polynomial of the first kind
+struct ChebyshevBasis{Int} <: TensorProductBasis
```

Comment:
```suggestion
struct ChebyshevBasis <: TensorProductBasis
```

It only needs to be parameterized if it's not concrete